pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -887,6 +895,18 @@ async def _responses_create(
         else:
             tool_choice = 'auto'
 
+        previous_response_id: str | None = None
+        for message in reversed(messages):
+            # Instead of sending the full message history, get provider_response_id
+            # (openai-compatible) from the latest matching ModelResponse and
+            # pass it to the next ModelRequest as previous_response_id to preserve context.
+            # Since the full history isn't needed, only the latest message is kept.
+            if isinstance(message, ModelResponse) and message.model_name:
+                if self._model_name in message.model_name:
+                    previous_response_id = message.provider_response_id
+                    messages = [messages[-1]]
```

Comment:
If the matched `ModelResponse` was not the most recent one, we should include all the messages that came after it, not just the last one

Reply:
@GDaamn Hmm, I don't think that's common but it is indeed technically supported. Then maybe we should only send `previous_response_id` if all responses came from the matching model.